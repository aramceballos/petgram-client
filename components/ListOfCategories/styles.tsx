import styled from 'styled-components'

export const List = styled.ul`
  display: flex;
  overflow: scroll;
  width: 100%;
  border-top: 1px solid #dbdbdb;
  border-bottom: 1px solid #dbdbdb;
  padding: 16px 0;
  background-color: #fafafa;
  -ms-overflow-style: none;
  scrollbar-width: none;
  &::-webkit-scrollbar {
    display: none;
  }
`

export const Item = styled.li`
  padding: 0 8px;
`
