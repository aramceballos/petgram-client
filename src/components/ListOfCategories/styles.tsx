import styled from 'styled-components'

export const List = styled.ul`
  display: flex;
  overflow: scroll;
  width: 100%;
  border: 1px solid #dbdbdb;
  margin-bottom: 24px;
  padding: 16px 0;
  border-radius: 3px;
`

export const Item = styled.li`
  padding: 0 8px;
`
