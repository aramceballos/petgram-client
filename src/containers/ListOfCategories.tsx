import React, { useEffect, useState } from 'react'

import ListOfCategoriesComponent from '../components/ListOfCategories'

const useCategoriesData = () => {
  const [categories, setCategories] = useState<ICategory[]>([])
  const [loading, setLoading] = useState(false)

  useEffect(() => {
    setLoading(true)
    fetch('http://localhost:5000/c')
      .then((res) => res.json())
      .then((response) => {
        setCategories(response.data)
        setLoading(false)
      })
  }, [])

  return { categories, loading }
}

const ListOfCategories = () => {
  const { categories, loading } = useCategoriesData()

  return <ListOfCategoriesComponent categories={categories} loading={loading} />
}

export default React.memo(ListOfCategories)
