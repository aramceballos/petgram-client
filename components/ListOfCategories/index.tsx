import React, { useEffect, useState } from 'react'
import Skeleton from '@material-ui/lab/Skeleton'

import Category from '../Category'
import { List, Item } from './styles'

type Props = {
  categories: ICategory[]
  loading?: boolean
}

const ListOfCategoriesComponent = ({ categories, loading }: Props) => {
  const [showFixed, setShowFixed] = useState(false)

  useEffect(() => {
    const onScroll = () => {
      const newShowFixed = window.scrollY > 200
      showFixed !== newShowFixed && setShowFixed(newShowFixed)
    }

    document.addEventListener('scroll', onScroll)

    return () => document.removeEventListener('scroll', onScroll)
  }, [showFixed])

  return (
    <List>
      {loading ? (
        <>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              animation="wave"
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              animation="wave"
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              animation="wave"
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              animation="wave"
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
        </>
      ) : (
        categories &&
        categories.map((category) => (
          <Item key={category.id}>
            <Category {...category} path={`/c/${category.id}`} />
          </Item>
        ))
      )}
    </List>
  )
}

export default ListOfCategoriesComponent
