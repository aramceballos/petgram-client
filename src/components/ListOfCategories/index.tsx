import React, { useEffect, useState } from 'react'
import Skeleton from '@material-ui/lab/Skeleton'

import Category from '../Category'
import { List, Item } from './styles'

type Props = {
  categories: ICategory[]
  loading: boolean
}

const ListOfCategoriesComponent = ({ categories, loading }: Props) => {
  const [showFixed, setShowFixed] = useState(false)

  useEffect(() => {
    const onScroll = () => {
      const newShowFixed = window.scrollY > 200
      showFixed !== newShowFixed && setShowFixed(newShowFixed)
    }

    document.addEventListener('scroll', onScroll)

    return () => document.removeEventListener('scroll', onScroll)
  }, [showFixed])

  return (
    <List>
      {loading ? (
        <>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
          <Item>
            <Skeleton
              style={{ marginBottom: '25px' }}
              variant="circle"
              width={68}
              height={68}
            />
          </Item>
        </>
      ) : (
        categories.map((category) => (
          <Item key={category.id}>
            <Category {...category} path={`/pet/${category.id}`} />
          </Item>
        ))
      )}
    </List>
  )
}

export default React.memo(ListOfCategoriesComponent)
