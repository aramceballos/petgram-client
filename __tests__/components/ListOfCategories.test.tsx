import React from 'react'
import { shallow } from 'enzyme'
import ListOfCategories from '../../components/ListOfCategories'

describe('<ListOfCategories />', () => {
  test('should render correctly with no data', () => {
    const props = {
      categories: [],
      loading: false,
    }
    const component = shallow(<ListOfCategories {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('should render correctly with data', () => {
    const props = {
      categories: [
        {
          id: 5,
          category: 'test_category',
          image_url: 'https://i.imgur.com/dJa0Hpl.jpg',
        },
        {
          id: 13,
          category: 'another_test',
          image_url: 'https://i.imgur.com/dJa0Hpl.jpg',
        },
      ],
      loading: false,
    }
    const component = shallow(<ListOfCategories {...props} />)
    expect(component).toMatchSnapshot()
  })
})
