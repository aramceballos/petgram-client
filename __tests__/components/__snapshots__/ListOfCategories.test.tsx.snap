// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfCategories /> should render correctly with data 1`] = `ShallowWrapper {}`;

exports[`<ListOfCategories /> should render correctly with no data 1`] = `ShallowWrapper {}`;
